src/main/java/com/faforever/api/featuredmods/FeaturedModFileEnricher.java

```diff
@@ -17,10 +29,22 @@ public void init(FafApiProperties fafApiProperties) {
   }
 
   @PostLoad
-  public void enhance(FeaturedModFile featuredModFile) {
+  public void enhance(FeaturedModFile featuredModFile) throws NoSuchAlgorithmException, InvalidKeyException {
     String folder = featuredModFile.getFolderName();
     String urlFormat = fafApiProperties.getFeaturedMod().getFileUrlFormat();
+    String secret = fafApiProperties.getFeaturedMod().getCloudflareHmacSecret();
+    long timeStamp = Instant.now().getEpochSecond();
+    URI featuredModUri = URI.create(String.format(urlFormat, folder, featuredModFile.getOriginalFileName()));
+
+    // Builds hmac token for cloudflare firewall verification as specified at
+    // https://support.cloudflare.com/hc/en-us/articles/115001376488-Configuring-Token-Authentication
+    Mac mac = Mac.getInstance(HMAC_SHA256);
+    mac.init(new SecretKeySpec(secret.getBytes(StandardCharsets.UTF_8), HMAC_SHA256));
+    byte[] macMessage = (featuredModUri.getPath() + timeStamp).getBytes(StandardCharsets.UTF_8);
+
+    String hmacEncoded = URLEncoder.encode(new String(Base64.getEncoder().encode(mac.doFinal(macMessage)), StandardCharsets.UTF_8), StandardCharsets.UTF_8);
+    String parameter = "%d-%s".formatted(timeStamp, hmacEncoded);
 
-    featuredModFile.setUrl(String.format(urlFormat, folder, featuredModFile.getOriginalFileName()));
+    featuredModFile.setUrl(UriComponentsBuilder.fromUri(featuredModUri).queryParam("verify", parameter).build().toString());
```

Comment:
You could just hardcode the query param in the featuredModUri similar to `faf.featured-mod.file-url-format`

Reply:
Unfortunately we can't because the URL path sans token is part of the token generation process

Reply:
Unless you mean the name of the param

Reply:
Yes e.g. `<url>?verify=%s`